docs/topics/jvm/java-interop.md

```diff
@@ -137,20 +144,76 @@ Set<@NotNull String> toSet(@NotNull Collection<@NotNull String> elements) { ...
 It leads to the following signature seen in Kotlin:
 
 ```kotlin
-fun toSet(elements: (Mutable)Collection<String>) : (Mutable)Set<String> { ... }
+fun toSet(elements: (Mutable)Collection<String>) : (Mutable)Set<String>
 ```
 
 Note the `@NotNull` annotations on `String` type arguments. Without them, you get platform types in the type arguments:
 
 ```kotlin
-fun toSet(elements: (Mutable)Collection<String!>) : (Mutable)Set<String!> { ... }
+fun toSet(elements: (Mutable)Collection<String!>) : (Mutable)Set<String!>
+```
+
+Kotlin also takes into account nullability annotations on type arguments of base classes and interfaces. For example,
+there are 2 Java classes with the signatures provided below.
+
+```java
+public class Base<T>
+```
+
+```java
+public class Derived extends Base<@Nullable String>
+```
+
+In the Kotlin code, passing the instance of `Derived` where the `Base<String>` assumed produces the warning.
+
+```kotlin
+fun takeBaseOfNotNullStrings(x: Base<String>) {}
+
+fun main() {
+    takeBaseOfNotNullStrings(Derived()) // warning 
```

Comment:
Please add details about the warning. If its message is short, you can just copy it. Otherwise, conduct its meaning in 3-5 words.

Reply:
Postponed to 1.5.20